pkg/didmethod/httpuniversal/resolver.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package httpuniversal
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	errors "golang.org/x/xerrors"
+)
+
+//HTTPClient abstracts away http client library
+type HTTPClient interface {
+	Get(url string) (body []byte, err error)
+}
+
+//StandardHTTPClient is based on http.Client
+type StandardHTTPClient struct {
+	client *http.Client
+}
+
+//Get implements HTTP Get operation on standard http client
+func (shc StandardHTTPClient) Get(url string) (body []byte, err error) {
+	resp, err := shc.client.Get(url)
+	if err != nil {
+		return nil, errors.Errorf("HTTP Get request failed: %w", err)
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Errorf("Failed to read response body: %w", err)
+	}
+
+	err = resp.Body.Close()
+	if err != nil {
+		return nil, errors.Errorf("Failed to close response body: %w", err)
+	}
+	return b, nil
+}
+
+// URLBuilderCreator is a factory of URL builder based on base url and did
+type URLBuilderCreator func(baseURL, did string) (string, error)
+
+// QueryParamURLBuilder provides a factory of URL builder using query parameters
+func QueryParamURLBuilder(paramName string) URLBuilderCreator {
+	return func(baseURL, did string) (string, error) {
+		baseURI, err := url.ParseRequestURI(baseURL)
+		if err != nil {
+			return "", errors.Errorf("Invalid base url: %w", err)
+		}
+
+		params := url.Values{}
+		params.Add(paramName, did)
```

Comment:
The DID should be supplied in the path rather than as a query parameter.
Example: https://uniresolver.io/1.0/identifiers/did:sov:WRfXPg8dantKVubE3HX8pw

Note: the host and base path need to be configurable via option. Examples:
- https://example.com/did:sov:WRfXPg8dantKVubE3HX8pw
- https://example.com/document/did:sidetree:WRfXPg8dantKVubE3HX8pw

are also valid server configurations.

Reply:
True. My next version follows the spec more strictly.